src/WebJobs.Script/Extensions/HttpRequestExtensions.cs

```diff
@@ -88,6 +90,24 @@ public static bool IsAppServiceInternalRequest(this HttpRequest request, IEnviro
             return !request.Headers.Keys.Contains(ScriptConstants.AntaresLogIdHeaderName);
         }
 
+        public static bool IsPlatformInternalRequest(this HttpRequest request, IEnvironment environment = null)
+        {
+            environment = GetEnvironment(request, environment);
+            if (!environment.IsAppService())
+            {
+                return false;
+            }
+
+            var header = request.Headers[ScriptConstants.AntaresPlatformInternal];
+            string value = header.FirstOrDefault();
+            return string.Compare(value, "true", StringComparison.OrdinalIgnoreCase) == 0;
+        }
+
+        private static IEnvironment GetEnvironment(HttpRequest request, IEnvironment environment = null)
+        {
+            return environment ?? request.HttpContext.RequestServices?.GetService<IEnvironment>() ?? SystemEnvironment.Instance;
```

Comment:
In writing tests, I discovered that any IEnvironment configured on the test host wasn't being used.